src/common/types/store-data/card-view-model.ts

```diff
@@ -0,0 +1,15 @@
+import { GuidanceLink } from '../../../scanner/rule-to-links-mappings';
+import { InstanceResultStatus, UnifiedResult } from './unified-data-interface';
+export type CardRuleResultStatus = InstanceResultStatus | 'inapplicable';
+export interface CardRuleResult {
+    id: string;
+    nodes: CardResult[];
+    description: string;
+    url: string;
+    guidance: GuidanceLink[];
+}
+export type CardRuleResultsByStatus = {
+    [key in CardRuleResultStatus]: CardRuleResult[];
+};
+export interface CardResult extends UnifiedResult {}
+export const AllRuleResultStatuses: CardRuleResultStatus[] = ['pass', 'fail', 'unknown', 'inapplicable'];
```

Comment:
Is there no way to iterate the type? It will be tough to remember to update this value if other values are added to the enum.

Reply:
Unfortunately, I don't believe there is. I don't suspect we'll be adding to this that frequently though.